camel-core/src/main/java/org/apache/camel/impl/verifier/ResultErrorHelper.java

```diff
@@ -63,16 +63,31 @@ private ResultErrorHelper() {
         final Set<String> keys = new HashSet<>(parameters.keySet());
 
         for (OptionsGroup group : groups) {
-            if (keys.containsAll(group.getOptions())) {
+            final Set<String> required = required(group.getOptions());
+
+            final ResultErrorBuilder builder = new ResultErrorBuilder()
+                .code(VerificationError.StandardCode.INCOMPLETE_PARAMETER_GROUP)
+                .detail(VerificationError.GroupAttribute.GROUP_NAME, group.getName())
+                .detail(VerificationError.GroupAttribute.GROUP_OPTIONS, String.join(",", parameters(group.getOptions())));
+
+            if (keys.containsAll(required)) {
                 // All the options of this group are found so we are good
-                return Collections.emptyList();
+
+                final Set<String> excluded = excluded(group.getOptions());
+
+                final Set<String> shouldBeExcluded = new HashSet<>(keys);
+                shouldBeExcluded.retainAll(excluded);
+
+                if (shouldBeExcluded.isEmpty()) {
+                    // None of the excluded properties is present, also good
+                    return Collections.emptyList();
+                }
+
+                shouldBeExcluded.forEach(builder::parameterKey);
+                verificationErrors.add(builder.build());
             } else {
-                ResultErrorBuilder builder = new ResultErrorBuilder()
-                    .code(VerificationError.StandardCode.INCOMPLETE_PARAMETER_GROUP)
-                    .detail(VerificationError.GroupAttribute.GROUP_NAME, group.getName())
-                    .detail(VerificationError.GroupAttribute.GROUP_OPTIONS, String.join(",", group.getOptions()));
 
-                for (String option : group.getOptions()) {
+                for (String option : required) {
                     if (!parameters.containsKey(option)) {
                         builder.parameterKey(option);
                     }
```

Comment:
Maybe also add the keys which are present but forbidden for this group in a second loop?